src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -34,6 +36,31 @@ private class CacheItem
             public TxOut[] OriginalOutputs;
         }
 
+        /// <summary>
+        /// Rewind data with its byte size that will be kept in the queue until it is persisted to a storage
+        /// </summary>
+        private class QueuedRewindData
+        {
+            /// <summary>Rewind data to be saved.</summary>
+            public RewindData RewindData { get; set; }
+
+            /// <summary>The size of the rewind data in bytes.</summary>
+            public long? DataSize { get; set; }
+
+            public static QueuedRewindData Create(RewindData rewindData)
+            {
+                long size;
+                using (var ms = new MemoryStream())
+                {
+                    var bitcoinStream = new BitcoinStream(ms, true);
+                    bitcoinStream.ReadWrite(rewindData);
+                    size = bitcoinStream.Serializing ? bitcoinStream.Counter.WrittenBytes : bitcoinStream.Counter.ReadBytes;
```

Comment:
You know you are writing here, so you don't really need this if

Reply:
Sure, but most likely we will remove all this logic anyway as per above.